lambdas/catalog-updater/index.js

```diff
@@ -49,40 +90,54 @@ function getSwaggerFile(file) {
  * {
  *   id: 'YOUR_USAGE_PLAN_ID',
  *   name: 'Free',
+ *   // throttle could be undefined
+ *   throttle: {
+ *      burstLimit: 5,
+ *      rateLimit: 0.33
+ *   },
+ *   // quota could be undefined
+ *   quota: {
+ *      limit : 100,
+ *      offset : 0,
+ *      // "DAY", "WEEK", or "MONTH"
+ *      period : "WEEK"
+ *   },
  *   apis: [{
  *     id: 'YOUR_API_ID',
+ *     stage: 'YOUR_STAGE_NAME'
  *     image: '/sam-logo.png',
- *     swagger: petStoreSwaggerDefinition
+ *     swagger: '{ "swagger": 2.0, ... }'
  *   }]
  * }
  * @param {Object} usagePlan the API GW CS representation of a usage plan
- * @param {Array} swaggerFiles array of swagger file bodies retrieved from s3
+ * @param {Array} swaggerFileReprs array of swagger file representations, each with a body and path parameter
  * @returns {Object} a 'catalog object' with id, name, and apis properties
  */
-function usagePlanToCatalogObject(usagePlan, swaggerFiles) {
+function usagePlanToCatalogObject(usagePlan, swaggerFileReprs) {
   let catalogObject = {
     id: usagePlan.id,
     name: usagePlan.name,
+    throttle: usagePlan.throttle,
+    quota: usagePlan.quota,
     apis: []
   }
 
   _.forEach(usagePlan.apiStages, (apiStage) => {
     let api = {}
 
-    _.chain(swaggerFiles)
-      .find((swaggerFile) => {
-        console.log(`swaggerFile: ${swaggerFile}`)
-        console.log(`apiId: ${apiStage.apiId}`)
-        console.log(`swaggerFile.indexOf(apiStage.apiId): ${swaggerFile.indexOf(apiStage.apiId)}`)
-        return swaggerFile.indexOf(apiStage.apiId) !== -1
+    _.chain(swaggerFileReprs)
+      .find((swaggerFileRepr) => {
+        return swaggerFileRepr.apiStageKey === `${apiStage.apiId}:${apiStage.stage}`
       })
-      .tap((swaggerFile) => {
-        if(swaggerFile) {
-          api.swagger = JSON.parse(swaggerFile)
+      .tap((swaggerFileRepr) => {
+        if(swaggerFileRepr) {
+          let swaggerOrError = _.attempt(JSON.parse, swaggerFileRepr.body)
+          // parse the JSON if it's JSON, otherwise, leave it as a string
+          _.isError(swaggerOrError) ? api.swagger = swaggerFileRepr.body : api.swagger = swaggerOrError
```

Comment:
Will this mean that when it gets sent to the front end, the spec will be a YAML string? If so, there's a change or two we'll need to make on the front-end. We attempt to grab data (currently just the API Name and Version) from the (JSON) swagger doc.

Reply:
Yep. I'm not sure that we have any option other than to send it stringified? For now, we'll continue cautioning users to only use JSON (but this brings YAML support to the backend). Speaking of which, there's a README change that needs to go with this.

Reply:
Only other option I can think of is converting the yaml. If we did it on the server side, wouldn't need the regex's to get apiId and stageName. (Or rather, we'd need different regexes.)

Warning is good enough for me. YAML support probably isn't a big deal while we only support API Gateway Swagger.